src/Particle/ParticleContainer.cpp

```diff
@@ -14,21 +14,66 @@ namespace
 {
     using namespace amrex;
 
-    // Note - this function MUST be consistent with AMReX_Particle.H
-    Long unpack_id (uint64_t idcpu) {
-        Long r = 0;
+    py::object pack_ids (py::array_t<uint64_t> idcpus,
+                         py::array_t<amrex::Long> ids)
+    {
+        if (idcpus.ndim() != 1) {
+            throw std::runtime_error("Input should be 1-D NumPy array");
+        }
+
+        auto buf = idcpus.request();
+        auto buf2 = ids.request();
+        if (buf.size != buf2.size) throw std::runtime_error("sizes do not match!");
+
+        int N = idcpus.shape()[0];
+        for (int i = 0; i < N; i++) {
+            uint64_t* idcpus_ptr = (uint64_t*) buf.ptr;
+            amrex::Long* ids_ptr = (amrex::Long*) buf2.ptr;
+            particle_impl::pack_id(idcpus_ptr[i], ids_ptr[i]);
+        }
+        return py::cast<py::none>(Py_None);
+    }
 
-        uint64_t sign = idcpu >> 63;  // extract leftmost sign bit
-        uint64_t val  = ((idcpu >> 24) & 0x7FFFFFFFFF);  // extract next 39 id bits
+    py::object pack_cpus (py::array_t<uint64_t> idcpus,
+                          py::array_t<int> cpus)
+    {
+        if (idcpus.ndim() != 1) {
+            throw std::runtime_error("Input should be 1-D NumPy array");
+        }
+
+        auto buf = idcpus.request();
+        auto buf2 = cpus.request();
+        if (buf.size != buf2.size) throw std::runtime_error("sizes do not match!");
```

Comment:
Shall we enclose this in parenthesis just for consistency with the other one-line if conditions?
```suggestion
        if (buf.size != buf2.size) {
            throw std::runtime_error("sizes do not match!");
        }
```